mtgjson5/price_builder.py

```diff
@@ -252,3 +297,159 @@ def build_prices(self) -> Tuple[Dict[str, Any], Dict[str, Any]]:
+
+        :return: Tuple of (archive_prices_v2, today_prices_v2)
+        """
+        LOGGER.info("Prices Build - Building Prices (V2 Format)")
+
+        # Get today's v2 price database
+        LOGGER.info("Building new v2 price data from providers")
+        today_prices_v2 = self.build_today_prices_v2()
+        if not today_prices_v2 or today_prices_v2.get_record_count() == 0:
+            LOGGER.warning("V2 pricing information failed to generate")
+            empty_v2 = MtgjsonPricesV2Container()
+            return empty_v2, empty_v2
+
+        if not MtgjsonConfig().has_section("Prices"):
+            return today_prices_v2, today_prices_v2
+
+        bucket_name = MtgjsonConfig().get("Prices", "bucket_name")
+        bucket_object_path_v2 = MtgjsonConfig().get(
+            "Prices", "bucket_object_path_v2", fallback="AllPricesv2.json.tar.xz"
+        )
+
+        # Download existing v2 archive (dict format: {provider: [records...]})
+        archive_dict = self.get_price_archive_data(bucket_name, bucket_object_path_v2)
+
+        # Reconstruct container from downloaded archive
+        archive_prices_v2 = self._reconstruct_v2_container_from_dict(archive_dict)
+
+        # Merge today's v2 data into archive
+        LOGGER.info("Merging new v2 price data into archive")
+        self._merge_v2_prices(archive_prices_v2, today_prices_v2)
+
+        # Prune old v2 records
+        LOGGER.info("Pruning v2 price archive")
+        self._prune_v2_prices_archive(archive_prices_v2)
+
+        # Write and upload v2 archive
+        LOGGER.info("Compressing v2 price data")
+        local_zip_file_v2 = constants.CACHE_PATH.joinpath(bucket_object_path_v2)
+        self.write_price_archive_data(local_zip_file_v2, archive_prices_v2.to_json())
+
+        LOGGER.info("Uploading v2 price data")
+        MtgjsonS3Handler().upload_file(
+            str(local_zip_file_v2), bucket_name, bucket_object_path_v2
+        )
+        local_zip_file_v2.unlink()
+
+        return archive_prices_v2, today_prices_v2
+
+    def _reconstruct_v2_container_from_dict(
+        self, archive_dict: Dict[str, Any]
+    ) -> MtgjsonPricesV2Container:
+        """
+        Reconstruct a MtgjsonPricesV2Container from archived dict format.
+        Archive format is {provider: [record_dicts...]}
+
+        :param archive_dict: Downloaded archive data
+        :return: Reconstructed container
+        """
+        from .classes.mtgjson_prices_v2 import MtgjsonPricesRecordV2
+
+        container = MtgjsonPricesV2Container()
+        if not archive_dict:
+            return container
+
+        LOGGER.info("Loading existing v2 archive data")
+        # Archive is {provider: [records...]} format
+        for provider, records_list in archive_dict.items():
+            for record_dict in records_list:
+                # Reconstruct MtgjsonPricesRecordV2 from dict
+                record = MtgjsonPricesRecordV2(
```

Comment:
This re-construction of an object doesn't feel very necessary. There's a few ways to do this, but I think we can skip this step.